### PR TITLE
SSPROD-10240-M replace ".." concatenation with table.concat

### PR DESCRIPTION
**Pull request by VadimZy**
**What type of PR is this?**

kind: design

**Any specific area of the project related to this PR?**

area: engine

area: rules

**What this PR does / why we need it**:
OOM error when parsing big Falco tuner files

**Which issue(s) this PR fixes**:
OOM error when parsing big Falco tuner files

Fixes #

**Special notes for your reviewer**:
The original code formating was not perfect - a mix of 2,3,4 indentations

Please use ignore whitespace when reviewing.

**Does this PR introduce a user-facing change?**:
"NONE" 

```release-note

```

### CODE REVIEW

File: userspace/engine/lua/rule_loader.lua [hunk 1]
```diff
@@ -766,65 +766,69 @@ end
...
+                -- If the corresponding operator is one that works on lists, possibly add surrounding parentheses.
+                if defined_list_comp_operators[comps[k]] then
+                    istr = paren_item(ival)
+                else
+                    -- Quote the value if not already quoted
+                    istr = quote_item(ival)
+                end
+            end
+
+            icond[#icond + 1] = fields[k] .. " " .. comps[k] .. " " .. istr
```
Comment (mstemm) [inline userspace/engine/lua/rule_loader.lua:816]: This one probably isn't as important, as it's just 3 big blocks of text per exception fields + values, but you might want to table concat here too.

File: userspace/engine/lua/rule_loader.lua [hunk 2]
```diff
@@ -766,65 +766,69 @@ end
...
 
-   icond = icond..")"
+    icond[#icond + 1] = ")"
 
-   -- Don't return a trivially empty condition string
-   if icond == "()" then
-      icond = ""
-   end
+    -- Don't return a trivially empty condition string
+    local ret = table.concat(icond)
```
Comment (mstemm) [inline userspace/engine/lua/rule_loader.lua:826]: I didn't double-check, but does concatenating directly with no spaces give you a meaningful condition snippet?

**Review by mstemm — CHANGES_REQUESTED** [review]

File: userspace/engine/lua/rule_loader.lua (hunk 1 above)
Comment (VadimZy) [reply userspace/engine/lua/rule_loader.lua:816]: I thought about it but then we will create a temp table for each iteration and we might have the same mem consumption. To clean up, we probably should take this piece in a separate func. In our tests, this was a problem anyway.

File: userspace/engine/lua/rule_loader.lua (hunk 2 above)
Comment (VadimZy) [reply userspace/engine/lua/rule_loader.lua:826]: Yes, the rules (exceptions) were correct.

File: userspace/engine/lua/rule_loader.lua [hunk 3]
```diff
@@ -766,65 +766,69 @@ end
...
+            end
+            local ival = values[k]
+            local istr = ""
+
+            -- If ival is a table, express it as (titem1, titem2, etc)
+            if type(ival) == "table" then
+                istr = "("
+                for _, item in ipairs(ival) do
+                    if istr ~= "(" then
+                        istr = istr .. ", "
```
[the remark this answers is not in the document]
Comment (VadimZy) [reply userspace/engine/lua/rule_loader.lua:801]: It was not a problem when I run tests, probably cuz `local istr = ""` local gets collected on each iteration

**Comment by VadimZy** [comment]
@mstemm @ivanlysiuk-sysdig closing this due to base divergence; new opened 
https://github.com/draios/oss-falco/pull/22

**Pull request closed without merging**